api/agent/protocol/factory.go

```diff
@@ -172,3 +172,57 @@ func New(p Protocol, in io.Writer, out io.Reader) ContainerIO {
 // IsStreamable says whether the given protocol can be used for streaming into
 // hot functions.
 func IsStreamable(p Protocol) bool { return New(p, nil, nil).IsStreamable() }
+
+type ClampWriter struct {
```

Comment:
`func_logger.go` has this guy too.

Reply:
we used to have `common/writers.go` that had various io.Writer utilities, we could do that again

Reply:
I should move these to common/writers.go. But the func_logger one is a bit specialized for log writing it seems.